src/twisted/names/client.py

```diff
@@ -439,6 +451,43 @@ def _cbLookupZone(self, result, connector):
 
 
 
+class IPv6Resolver(Resolver):
+    """
+    Selects only IPv6 resolv.conf entries.
+    """
+    def maybeAddNameserver(self, nameserver, servers):
+        resolver = (nativeString(nameserver), dns.PORT)
+        if b':' in nameserver:
```

Comment:
For something this simple I think the functions already in Twisted should be fine.  Separately, it might be a good idea to replace their _implementations_ with the `ipaddress` module.